.github/workflows/docs.yml

```diff
@@ -0,0 +1,29 @@
+name: docs
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ '*' ]
```

Comment:
We probably don't want to run this on PRs. Doing so could generate docs for bound to change features (say we iterate several times over a new API and no release includes it).

Reply:
Makes sense. Adjustment made.